src/goto-synthesizer/enumerative_loop_contracts_synthesizer.cpp

```diff
@@ -100,13 +101,13 @@ void enumerative_loop_contracts_synthesizert::init_candidates()
       // we only synthesize invariants and assigns for unannotated loops
       if(loop_end->condition().find(ID_C_spec_loop_invariant).is_nil())
       {
-        // Store the loop guard.
-        exprt guard =
-          get_loop_head(
-            loop_end->loop_number,
-            goto_model.goto_functions.function_map[function_p.first])
-            ->condition();
-        neg_guards[new_id] = guard;
+        // Store the loop guard if exists.
+        auto loop_head = get_loop_head(
+          loop_end->loop_number,
+          goto_model.goto_functions.function_map[function_p.first]);
+
+        if(loop_head->has_condition())
+          neg_guards[new_id] = loop_head->condition();
```

Comment:
I am admittedly surprised: are there cases where the loop head has no condition (`true_exprt()` also is a condition)?

Reply:
Seems like goto conversion would also produce this for other input sources. Ok, glad to have learned something :-)